### ISSUE TITLE
Slowness in makeAggregateBom command

### ISSUE THREAD

**Issue by justinabrahms** [CycloneDX/cyclonedx-maven-plugin#262]
Hey folks.

I'm generating both spdx and cyclonedx sboms in my build. The SPDX maven plugin takes 2-3 seconds. This plugin takes 20-30 seconds. This is a general "Can we please fix this so that generation happens faster?"

I'm happy to provide any data you need. My maven config is just copied straight from your readme. Build output here:

```
15:43:28 [INFO] --- cyclonedx-maven-plugin:2.7.1:makeAggregateBom (default) @ myapp ---
15:43:29 [INFO] CycloneDX: Parameters
15:43:29 [INFO] ------------------------------------------------------------------------
15:43:29 [INFO] schemaVersion          : 1.3
15:43:29 [INFO] includeBomSerialNumber : true
15:43:29 [INFO] includeCompileScope    : true
15:43:29 [INFO] includeProvidedScope   : true
15:43:29 [INFO] includeRuntimeScope    : true
15:43:29 [INFO] includeTestScope       : false
15:43:29 [INFO] includeSystemScope     : true
15:43:29 [INFO] includeLicenseText     : false
15:43:29 [INFO] outputReactorProjects  : true
15:43:29 [INFO] outputFormat           : all
15:43:29 [INFO] outputName             : bom
15:43:29 [INFO] ------------------------------------------------------------------------
15:43:50 [INFO] CycloneDX: Creating BOM
15:43:50 [INFO] CycloneDX: Writing BOM (XML): /home/jenkins/workspace/myapp/target/bom.xml
15:43:50 [INFO] CycloneDX: Validating BOM (XML): /home/jenkins/workspace/myapp/target/bom.xml
15:43:51 [INFO] CycloneDX: Writing BOM (JSON): /home/jenkins/workspace/myapp/target/bom.json
15:43:51 [INFO] CycloneDX: Validating BOM (JSON): /home/jenkins/workspace/myapp/target/bom.json
```

**Comment by stevespringett** [comment]
You may want to clarify, but the last time I ran the SPDX Maven plugin, it omitted all transitive dependencies. If this is still the case, it would certainly explain why it doesn't take long. Also, I don't think it supports multi-module aggregation.

BTW, there were tremendous performance improvements in CycloneDX Maven Plugin v2.7.4, so you may want to upgrade to that release if you can.

**Comment by hboutemy** [comment]
I suppose you are running in a multi-module project, then what takes time is not one execution of the goal, but the fact that it executes n*n times (n being the number of modules)

if that's the case, this has been fixed in 2.7.4 #242 : please test and confirm

**Comment by hboutemy** [comment]
no feedback, closing the issue as fixed
